azure-mgmt-dns/src/main/java/com/microsoft/azure/management/dns/implementation/ARecordSetImpl.java

```diff
@@ -21,15 +21,13 @@
 class ARecordSetImpl
         extends DnsRecordSetImpl
         implements ARecordSet {
-    ARecordSetImpl(final DnsZoneImpl parent, final RecordSetInner innerModel) {
-        super(parent, innerModel);
+    ARecordSetImpl(final String name, final String type, final DnsZoneImpl parent, final RecordSetInner innerModel) {
```

Comment:
Do we need the second param `final String type` for each specialized record-set?, given each impl know (e.g. ARecordSetImpl) what type it is. I mean, was thing like

```java
ARecordSetImpl(final String name, final DnsZoneImpl parent, final RecordSetInner innerModel) { 
    super(name, RecordType.A.toString(), parent, innerModel); 
}
```